### ISSUE TITLE
e2e flake: Port forwarding With a server that expects a client request [It] should support a client that connects, sends data, and disconnects [Conformance] 

### ISSUE THREAD

**Issue by derekwaynecarr** [kubernetes/kubernetes#21990]
```
[Fail] Port forwarding With a server that expects a client request [It] should support a client that connects, sends data, and disconnects [Conformance] 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:163

â€¢ Failure [311.018 seconds]
Port forwarding
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:268
  With a server that expects a client request
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:216
    should support a client that connects, sends data, and disconnects [Conformance] [It]
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:215

    Feb 25 08:07:11.060: Pod did not start running: gave up waiting for pod 'pfpod' to be 'running' after 5m0s

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/portforward.go:163
```

**Comment by lavalamp** [comment]
I guess CSI sort of owns this test, even though the failure doesn't sound like it's something we own.

**Comment by lavalamp** [comment]
@derekwaynecarr Can you post the link of where you got that log message?

And then @krousey, can you dig through the logs to figure out what happened to the pod?

**Comment by lavalamp** [comment]
Pods not starting is a theme today: #21940

**Comment by derekwaynecarr** [comment]
I accidentally deleted the comment.  Apologies.

**Comment by krousey** [comment]
@derekwaynecarr So we don't know where the logs are?

**Comment by derekwaynecarr** [comment]
It came on the referenced PR in this issue, but I have been getting multiple flakes, and accidentally deleted the comment so I don't know how to find the list of all build logs ever associated with #20446

**Comment by derekwaynecarr** [comment]
@krousey - found the link in history: 

https://doc-02o22-1koto-s-googleusercontent.commondatastorage.googleapis.com/gs/rsbp87i9depc1vki877t9urscr6pcu7m/92kjeqi7cg1v4qhmj1mlpd9citr9blb7/1456416000000/kubernetes-jenkins/01814200508253236101/ChJrdWJlcm5ldGVzLWplbmtpbnMSSXByLWxvZ3MvcHVsbC8yMDQ0Ni9rdWJlcm5ldGVzLXB1bGwtYnVpbGQtdGVzdC1lMmUtZ2NlLzMwNDU0L2J1aWxkLWxvZy50eHQ?a=AGjQbs7Ovdl2rzpO0XBK37fTGulIPKq4_wwoaU_GMJzj6W6EF5sAi37PJwGws4ysME00Yz4oAnsncuyR-VcmvXE0Uyg7u2jmorCAgqBfji4YNS5GLdGEoVG5V4lYoL_5C6eAyQQq5DEos4IxfkVAhnICEJU_14053_5rDsecX2xLwBlMFl2j4Ovr-9GshVacwbgI4ykz3Nf9NbVArh3TS1afv3gUL9LWQRJyUDNGfw9lzAEVIZzoX3Lq5hWEAVbRBngGEKOjYEjg3L6eo3mGOfzxItxJUJsIKmuJh-BrQ-oSBgtoistnwekmS3yXv1_doNXxZpkSA0oVwlaqiUtEuPo-mt45SqsCqk-HZ5hKGLzsxvwe0Jvp_f-I5AHz_Af5kR_Z5wd5ew5lLyP32iTZCwLWuupKyrgFv3TrAVPRDutgx6Tuis2vo9M_pB4Q4-NF0PhYPTIdGmggXhXlAr-yWFJ5FF8bU2MMes5YHudsZ-2NhpgK-4Nj1kMe0exxcxbv0C-XTYJ9kbgvp11wN6AVto6M6wIKmZ--ctiQnBqS12vKHD5eozZkHgq8N4jW1iNpneg3uaTu5Homg9wkwQlTdhgh5XL5LIjy0c8CnuXzxWLFw3rEn8XwAxjTNk8HTe_32bXtJ48zOXyEYtbkjMh0qcIciYlaVI8o-0sa7X1ZnPR9tjEBfAPa-WcGZazA1e4w5Y_D8LVbVMdkOvmO0KHZOw7RS3Lek4BfrqpY7WLfcjTwN_ZX_dn0lqf8UvEVcY-KE0BO6ROFYlA6pdNUS2pSJVkL5uRadJTibg

**Comment by ncdc** [comment]
I think we need more of a generic "pods failed to start for 5m" issue instead of creating a new issue for each specific test case when this happens

**Comment by krousey** [comment]
@derekwaynecarr That link doesn't work for me. After some sleuthing, I think it's this

https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/20446/kubernetes-pull-build-test-e2e-gce/30454/

**Comment by krousey** [comment]
Found this in the docker log:

```
time="2016-02-25T16:02:09.864877588Z" level=error msg="Handler for GET /images/gcr.io/google_containers/portforwardtester:1.0/json returned error: No such image: gcr.io/google_containers/portforwardtester:1.0" 
```

**Comment by krousey** [comment]
Looks like gcr.io was returning a lot of 404s

```
time="2016-02-25T16:02:08.954719584Z" level=error msg="Handler for GET /images/gcr.io/google_containers/busybox:1.24/json returned error: No such image: gcr.io/google_containers/busybox:1.24" 
time="2016-02-25T16:02:08.954763054Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/busybox:1.24" statusCode=404 
time="2016-02-25T16:02:09.864877588Z" level=error msg="Handler for GET /images/gcr.io/google_containers/portforwardtester:1.0/json returned error: No such image: gcr.io/google_containers/portforwardtester:1.0" 
time="2016-02-25T16:02:09.864922542Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/portforwardtester:1.0" statusCode=404 
time="2016-02-25T16:02:10.853391872Z" level=error msg="Handler for POST /containers/c068fb9168277d5391f1dc9e65e3951ddcfe5ed8ebc73f3e15801ebaeda4b76c/stop returned error: Container already stopped" 
time="2016-02-25T16:02:10.853432587Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:11.015460305Z" level=error msg="Handler for GET /images/redis/json returned error: No such image: redis" 
time="2016-02-25T16:02:11.015502712Z" level=error msg="HTTP Error" err="No such image: redis" statusCode=404 
time="2016-02-25T16:02:14.844816654Z" level=error msg="Handler for GET /images/gcr.io/google_containers/mounttest:0.6/json returned error: No such image: gcr.io/google_containers/mounttest:0.6" 
time="2016-02-25T16:02:14.844881797Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/mounttest:0.6" statusCode=404 
time="2016-02-25T16:02:16.404734207Z" level=error msg="Handler for GET /images/gcr.io/google_containers/nettest:1.7/json returned error: No such image: gcr.io/google_containers/nettest:1.7" 
time="2016-02-25T16:02:16.404799329Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/nettest:1.7" statusCode=404 
time="2016-02-25T16:02:18.861556857Z" level=error msg="Handler for GET /images/gcr.io/google_containers/test-webserver/json returned error: No such image: gcr.io/google_containers/test-webserver" 
time="2016-02-25T16:02:18.861620707Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/test-webserver" statusCode=404 
time="2016-02-25T16:02:19.808133643Z" level=error msg="Handler for GET /images/gcr.io/google_containers/serve_hostname:1.1/json returned error: No such image: gcr.io/google_containers/serve_hostname:1.1" 
time="2016-02-25T16:02:19.808193884Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/serve_hostname:1.1" statusCode=404 
time="2016-02-25T16:02:25.000457835Z" level=error msg="Handler for GET /images/gcr.io/google_containers/eptest:0.1/json returned error: No such image: gcr.io/google_containers/eptest:0.1" 
time="2016-02-25T16:02:25.000511929Z" level=error msg="HTTP Error" err="No such image: gcr.io/google_containers/eptest:0.1" statusCode=404 
time="2016-02-25T16:02:28.491723163Z" level=error msg="Handler for POST /containers/556a7fc71efd274236114a38ca23cb7032df0c2b8af6fe21ce776ba0298ba35e/stop returned error: Container already stopped" 
time="2016-02-25T16:02:28.491800687Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:29.076101378Z" level=error msg="Handler for GET /images/nginx/json returned error: No such image: nginx" 
time="2016-02-25T16:02:29.076172586Z" level=error msg="HTTP Error" err="No such image: nginx" statusCode=404 
time="2016-02-25T16:02:30.872061116Z" level=error msg="Handler for POST /containers/dabc52cf80d845147013727926933842422ee842c2fa01085b8130d5fba6bf2d/stop returned error: Container already stopped" 
time="2016-02-25T16:02:30.872116744Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:33.688102211Z" level=error msg="Handler for POST /containers/5dfadac769e98639a8f5b3660dd35d3d5d5b9627e179d66b1b0e51a01c2ed36c/stop returned error: Container already stopped" 
time="2016-02-25T16:02:33.688173599Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:33.691236807Z" level=error msg="Handler for POST /containers/b0d91d0bc232e835fdd4985beab96406797b0acc535bde030c8de3519a8ff439/stop returned error: Container already stopped" 
time="2016-02-25T16:02:33.691282437Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:34.858249627Z" level=error msg="Handler for POST /containers/5dfadac769e98639a8f5b3660dd35d3d5d5b9627e179d66b1b0e51a01c2ed36c/stop returned error: Container already stopped" 
time="2016-02-25T16:02:34.858307271Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:34.862532133Z" level=error msg="Handler for POST /containers/b0d91d0bc232e835fdd4985beab96406797b0acc535bde030c8de3519a8ff439/stop returned error: Container already stopped" 
time="2016-02-25T16:02:34.862580147Z" level=error msg="HTTP Error" err="Container already stopped" statusCode=304 
time="2016-02-25T16:02:53.049006799Z" level=error msg="Handler for GET /images/busybox/json returned error: No such image: busybox" 
time="2016-02-25T16:02:53.049065607Z" level=error msg="HTTP Error" err="No such image: busybox" statusCode=404 
```

**Comment by ncdc** [comment]
I don't think that's necessarily indicative of an actual problem. I'm
pretty sure it shows up in successful runs (for various images).

**Comment by ncdc** [comment]
Ok, the 404s would be an issue (I didn't see that comment when I was replying via email)

**Comment by krousey** [comment]
I'm unsure how to proceed on this. Do we just accept that gcr.io flaked on us? How do we fix it?

**Comment by ncdc** [comment]
If we have an external dependency and there's an outage that's is out of
our control, there's nothing we can do to fix it. Even if we increase the
timeout, at some point, we'll hit it again due to an extended outage
somewhere.

It would be ideal if we could automatically identify test "failures" as
transient due to external outages. If that's not possible, the next best
thing I can think of would be an easy way to mark one of these as a
transient failure, so that (in the absence of other issues) we can unblock
the merge queue.

**Comment by mml** [comment]
I've been experiencing this repeatedly for several days trying to start an e2e cluster.  At first I thought it was due to #21523, but now I'm not sure.  I'll see if I can find out what's going on with the google container registry.

**Comment by krousey** [comment]
re-assigning to @mml while I'm out

**Comment by dims** [comment]
Duplicate of https://github.com/kubernetes/kubernetes/issues/21824 ?

**Comment by mml** [comment]
I never did anything with this back in March.  @krousey is in a better position to dup.
